src/Sarif/NotYetAutoGenerated/ReportingConfiguration.cs

```diff
@@ -47,9 +47,9 @@ public SarifNodeKind SarifNodeKind
         /// <summary>
         /// Specifies the failure level for the report.
         /// </summary>
-        [DataMember(Name = "level", IsRequired = false, EmitDefaultValue = false)]
+        [DataMember(Name = "level", IsRequired = false, EmitDefaultValue = true)]
         [DefaultValue(FailureLevel.Warning)]
-        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Populate)]
```

Comment:
>[JsonProperty(DefaultValueHandling = DefaultValueHandling.Populate)] [](http://example.com/codeflow?start=8&length=68)

(Hello! I am newly assigned to take a look at the level default value issue. I might miss some context)

just for my unsersatnding for this line only, if we are going this route this should be set to DefaultValueHandling.Include not DefaultValueHandling.Populate right, 
which eaquals to remove this line 
since it is the default of DefaultValueHandling attribute.

Reply:
There are a few issues at play but this proposed PR only resolved one of them. 

1. The default value is warning. This isn't compliant with the spec. I didn't fix that here. 
2. When serializing, the default value isn't written.

I chose populate instead of include in case there were users of the sarif ask that had serialized out some files where the levels got skipped - when reading them back in would repopulate them back to warning. The downside of course being 1. That for any file not from the C# sarif sdk a blank level does not mean warning. However, changing this behavior is sort of a breaking change, so I wasn't sure if we wanted to do it at this point. 

I think a more holistic solution is as you propose to use include instead of populate and fix the default value behavior to comply with the spec.

Reply:
thanks, I created a PR, just for discussion, can not find you in the review user list to add you.
[https://github.com/microsoft/sarif-sdk/pull/2399](https://github.com/microsoft/sarif-sdk/pull/2399)